src/lib/operators/validate.cpp

```diff
@@ -36,6 +37,17 @@ bool Validate::is_row_visible(TransactionID our_tid, CommitID snapshot_commit_id
   return snapshot_commit_id < end_cid && ((snapshot_commit_id >= begin_cid) != (row_tid == our_tid));
 }
 
+bool Validate::is_entire_chunk_visible(const std::shared_ptr<const Chunk>& chunk, const CommitID snapshot_commit_id) {
+  DebugAssert(!std::dynamic_pointer_cast<const ReferenceSegment>(chunk->get_segment(ColumnID{0})),
+    "is_entire_chunk_visible cannot be called on reference chunks.");
+
+  const auto mvcc_data = chunk->mvcc_data();
```

Comment:
Personally, I prefer `const auto&` for cases like this. If `mvcc_data()` ever returns a reference, we are on the safe side.